src/edu/stanford/nlp/sempre/Rule.java

```diff
@@ -127,6 +127,23 @@ public boolean isAnchored() {
     else if (a == 0.0)
       return false;
     else
-      return f == 1.0 ? false : !FloatingParser.opts.defaultIsFloating;
+      return f == 1.0 ? false : !Parser.opts.defaultIsFloating;
   }
+
+  public boolean isSegmentRoot() {
```

Comment:
Better to not add these here or else Rule.java is going to get bloated.  Add them as static methods whereever they are actually used.